x/cronos/keeper/msg_server.go

```diff
@@ -15,3 +22,139 @@ func NewMsgServerImpl(keeper Keeper) types.MsgServer {
 }
 
 var _ types.MsgServer = msgServer{}
+
+func (k msgServer) ConvertTokens(goCtx context.Context, msg *types.MsgConvertTokens) (*types.MsgConvertResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	acc, err := sdk.AccAddressFromBech32(msg.Address)
+	if err != nil {
+		return nil, err
+	}
+
+	params := k.GetParams(ctx)
+	evmParams := k.GetEvmParams(ctx)
+
+	for _, c := range msg.Amount {
+		switch c.Denom {
+		case params.IbcCroDenom:
+			if params.IbcCroDenom == "" {
+				return nil, sdkerrors.Wrap(types.ErrIbcCroDenomEmpty, "ibc is disabled")
+			}
+
+			// Send ibc token escrow address
+			err = k.bankKeeper.SendCoinsFromAccountToModule(ctx, acc, types.ModuleName, sdk.NewCoins(c))
+			if err != nil {
+				return nil, err
+			}
+			// Compute new amount, because basecro is a 8 decimals token, we need to multiply by 10^10 to make it
+			// a 18 decimals token
+			ten := big.NewInt(10)
+			exponent := ten.Exp(ten, ten, nil)
+			newAmount := sdk.NewCoin(evmParams.EvmDenom, c.Amount.Mul(sdk.NewIntFromBigInt(exponent)))
+
+			// Mint new coins
+			if err := k.bankKeeper.MintCoins(
+				ctx, types.ModuleName, sdk.NewCoins(newAmount),
+			); err != nil {
+				return nil, err
+			}
+
+			// Send Evm coins to receiver
+			if err := k.bankKeeper.SendCoinsFromModuleToAccount(
+				ctx, types.ModuleName, acc, sdk.NewCoins(newAmount),
+			); err != nil {
+				return nil, err
+			}
+
+		default:
+			if err := k.IsConvertEnabledCoins(ctx, msg.Amount...); err != nil {
+				return nil, err
+			}
+			// TODO wrap to erc20 tokens
```

Comment:
Maybe we can do this in different PR, to keep this one simpler?

Reply:
okay, I just finished this implementation but I added some todo

Reply:
Can you remove the unimplemented part from this PR?

Reply:
done